pallets/vesting/src/lib.rs

```diff
@@ -91,12 +94,20 @@ pub mod pallet {
 	pub type VestingSchedule<T: Config> =
 		StorageMap<_, Twox64Concat, UnixTime, Vec<(AccountOf<T>, BalanceOf<T>)>, OptionQuery>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn total_unvested_allocation)]
+	pub type TotalUnvestedAllocation<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
 		fn on_initialize(_: T::BlockNumber) -> Weight {
 			let vest_count = Self::vest();
 			<T as Config>::WeightInfo::vest(vest_count)
 		}
+
+		fn on_runtime_upgrade() -> Weight {
+			migrations::set_total_unvested_allocation::<T>()
```

Comment:
do you have to gate with a storage version and/or update the storage version for next time? e.g. https://github.com/OAK-Foundation/OAK-blockchain/blob/master/pallets/automation-time/src/lib.rs#L69

we don't have to, but we'd just have to be pretty careful about removing afterwards